src/server/worker/pipeline/common.go

```diff
@@ -63,100 +220,43 @@ func ReceiveSpout(
+			var buff bytes.Buffer
+			// copy pipe contents to buffer
+			n, err := io.Copy(&buff, out)
+			if err != nil {
+				logger.Logf("Received an EOF", err)
+				break
 			}
-		}()
-		for {
-			// this extra closure is so that we can scope the defer for FinishCommit
-
-			if err := func() (retErr2 error) {
-
-				// create a new tar reader
-				outTar := tar.NewReader(out)
-
-				var commit *pfs.Commit
-
-				// this loops through all the files in the tar that we've read from /pfs/out
-				for {
-					fileHeader, err := outTar.Next()
-					if errors.Is(err, io.EOF) {
-						err = openAndWait()
-						if err != nil {
-							return err
-						}
-						break
-					}
-					if err != nil {
-						return err
-					}
-					if commit == nil {
-						// start commit
-						commit, err = pachClient.PfsAPIClient.StartCommit(ctx, &pfs.StartCommitRequest{
-							Parent:     client.NewCommit(repo, ""),
-							Branch:     pipelineInfo.OutputBranch,
-							Provenance: []*pfs.CommitProvenance{client.NewCommitProvenance(ppsconsts.SpecRepo, repo, pipelineInfo.SpecCommit.ID)},
-						})
-						if err != nil {
-							return err
-						}
-
-						// finish the commit even if there was an issue
-						defer func() {
-							if err := pachClient.FinishCommit(repo, commit.ID); err != nil && retErr2 == nil {
-								// this lets us pass the error through if FinishCommit fails
-								retErr2 = err
-							}
-						}()
-					}
-					// put files into pachyderm
-					if pipelineInfo.Spout.Marker != "" && strings.HasPrefix(path.Clean(fileHeader.Name), pipelineInfo.Spout.Marker) {
-						// we'll check that this is the latest version of the spout, and then commit to it
-						// we need to do this atomically because we otherwise might hit a subtle race condition
-
-						// check to see if this spout is the latest version of this spout by seeing if its spec commit has any children
-						spec, err := pachClient.InspectCommit(ppsconsts.SpecRepo, pipelineInfo.SpecCommit.ID)
-						if err != nil && !errutil.IsNotFoundError(err) {
-							return err
-						}
-						if spec != nil && len(spec.ChildCommits) != 0 {
-							return errors.New("outdated spout, now shutting down")
-						}
-						_, err = pachClient.PutFileOverwrite(repo, ppsconsts.SpoutMarkerBranch, fileHeader.Name, outTar, 0)
-						if err != nil {
-							return err
-						}
-
-					} else if pipelineInfo.Spout.Overwrite {
-						_, err = pachClient.PutFileOverwrite(repo, commit.ID, fileHeader.Name, outTar, 0)
-						if err != nil {
-							return err
-						}
-					} else {
-						_, err = pachClient.PutFile(repo, commit.ID, fileHeader.Name, outTar)
-						if err != nil {
-							return err
-						}
-					}
-				}
-				return nil
-			}(); err != nil {
-				return err
+			if n == 0 {
+				continue
 			}
+			logger.Logf("Got data: ", n, buff.Len())
+			// send the contents to the tar reader
+			TarReader(buff, logger)
+			// and process any new files received
+			logger.Logf("Total files", fileCount)
```

Comment:
We should change the log messages to be more meaningful. I had them to debug the loop.